clayui.com/src/components/IconSearch/IconSearch.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+
+import icons from './icons.json';
+
+const spritemap = '/images/icons/icons.svg';
+
+export default () => {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [filteredIconNames, setFilteredIconNames] = useState([]);
+
+	const handleIconSearch = event => {
+		setSearchQuery(event.target.value);
+	};
+
+	useEffect(() => {
+		const filteredIcons = icons.filter(icon => {
+			if (icon.aliases.includes(searchQuery)) {
+				return icon.name;
+			}
+		});
+
+		setFilteredIconNames(
+			filteredIcons.map(icon => {
+				return icon.name;
+			})
+		);
+	}, [searchQuery]);
+
+	return (
+		<>
+			<ClayForm.Group>
+				<label htmlFor="iconSearch">{'Search Icons'}</label>
+
+				<ClayInput
+					id="iconSearch"
+					onChange={handleIconSearch}
+					placeholder="Search icons"
+					type="text"
+					value={searchQuery}
+				/>
+			</ClayForm.Group>
+
+			<ul className="lexicon-icon-list list-unstyled">
+				{filteredIconNames.length
+					? filteredIconNames.map((iconName, i) => (
+							<li key={i}>
+								<ClayIcon
+									spritemap={spritemap}
+									symbol={iconName}
+								/>
+
+								<span>{iconName}</span>
+							</li>
+					  ))
+					: icons.map((icon, i) => (
```

Comment:
Instead of rendering all of the icons if no results are found, we should show an empty state message like `No results found for "test"`

Reply:
For example, if you search for "zzzz" it shows all icons which is a little confusing for the end user.

![Screen Shot 2020-02-03 at 10 41 17 AM](https://user-images.githubusercontent.com/6843530/73684927-0447f300-467a-11ea-83dd-5d9c28444799.png)